### ISSUE TITLE
Communities and Spaces

### ISSUE THREAD

**Issue by agowa338** [mastodon/mastodon#19904]
### Pitch

I'd like to have communities and spaces in mastodon. Basically custom timelines that one can subscribe to and post to.
When making a post users should be able to select that community and the post should not be visible on the poster's profile nor on the local timeline. But only on the community-shared timeline.
The moderation should be done by the community admin instead of the instance admins.
Communities should be able to be public and private. Subscribing should be either open for anyone or approval based (choosable by the community admins)
A public community should be viewable by everyone. But only if they actively choose that community. Not within any other feed.
In a private community only subscribed members of said community should be able to view the posts.

Tl;Dr: communities should provide a way to get something similar to a "local timeline" without having an account on the same instance.

### Motivation

Migrating from twitter communities to mastodon. Just setting up an instance has the issue that many already have accounts on other instances and don't want to create multiple accounts with multiple different timelines, but still participate in a community with shared experiences.
Example https://twitter.com/BrownEnvmgmt/status/1588949136740876288?s=20, https://twitter.com/ChistinaHartman/status/1588962562733117442?s=20, ...

**Comment by ClearlyClaire** [comment]
Groups are already planned and mostly implemented in a feature branch, but we are currently focusing on releasing 4.0 and dealing with the sudden influx of users right now.

Hopefully the group implementations we have worked on should be good enough to have groups (albeit necessarily public for now), but we may have to revisit aspects due to potential scaling issues.

**Comment by ClearlyClaire** [comment]
It has already been discussed at length with other Fediverse projects, although no widespread consensus has been achieved. There's also reasons why we are explicitly incompatible with some other projects in that matter.
